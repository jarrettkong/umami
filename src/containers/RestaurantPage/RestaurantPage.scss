.RestaurantPage {
	background-color: white;
	padding: 25px;
	min-height: calc(100vh - 60px);
	& h2 {
		margin-top: 0;
	}

	& ul {
		list-style: none;
		padding: 0;
		border-bottom: 1px solid lightgrey;

		& li {
			padding: 20px;
			border-top: 1px solid lightgrey;

			& a {
				text-decoration: none;
				color: inherit
			}
		}
	}
}

.RestaurantPage-review-text {
	font-style: italic;
	font-weight: 300;
}

.yelp-link {
	background: url('../../util/images/yelp-logo.png');
	background-color: #d32323;
	background-size: cover;
	background-position: center;
	color: white;
	font-weight: 600;
	padding: 15px 50px;
	border-radius: 5px;
	position: absolute;
	text-decoration: none;
	bottom: 0;
	left: 0;
}

.RestaurantPage-flex-wrapper {
	display: flex;
	justify-content: flex-start;

	& img {
		height: 400px;
		width: 400px;
		object-fit: cover;
		object-position: center;
	}

	& > div {
		margin-left: 50px;
		position: relative;
	}
}
