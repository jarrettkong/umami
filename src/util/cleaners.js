export const cleanDetails = details => {
	return {
		id: details.id,
		name: details.name,
		address: details.location.address1,
		city: `${details.location.city}, ${details.location.state}`,
		image: details.image_url,
		yelpUrl: details.url,
		phone: details.display_phone,
		rating: details.rating,
		reviewCount: details.review_count,
		categories: details.categories.map(c => c.title),
		coordinates: details.coordinates,
		photos: details.photos,
		price: details.price || 'No price information available'
	};
};

export const cleanReviews = (reviews, id) => {
	return {
		id,
		reviews: reviews.map(r => {
			return {
				rating: r.rating,
				name: r.user.name,
				text: r.text,
				date: r.time_created.slice(0, 10),
				url: r.url
			};
		})
	};
};
