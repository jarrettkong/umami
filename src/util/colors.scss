$main-color: #2a5998;
$background-color: #fcfcfc;
// $main-color: #c90000;
$body-copy-color: #393939;
$body-copy-font: 'Roboto';
