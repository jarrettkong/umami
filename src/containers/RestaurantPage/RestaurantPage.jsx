import React, { Component } from 'react';
import Loader from '../../components/Loader/Loader';
import { connect } from 'react-redux';
import { addDetails, addReviews } from '../../actions';
import { cleanDetails, cleanReviews } from '../../util/cleaners';
import Yelp from '../../api/Yelp';
import StarRatings from 'react-star-ratings'

export class RestaurantPage extends Component {
	state = {
		loading: false,
		details: {},
		reviews: [],
		error: null
	};

	componentDidMount() {
		const details = this.props.details.find(r => r.id === this.props.id);
		if (!details) {
			this.setState({ loading: true }, async () => {
				await this.getRestaurantDetails(this.props.id);
				this.setState({ loading: false, error: null });
			});
		} else {
			const reviews = this.props.reviews.find(r => r.id === this.props.id);
			this.setState({ details, reviews: reviews.reviews });
		}
	}

	getRestaurantDetails = async id => {
		try {
			const detailsRes = await Yelp.get(`/businesses/${id}`);
			const reviewsRes = await Yelp.get(`/businesses/${id}/reviews`);
			const details = cleanDetails(detailsRes.data);
			const reviews = cleanReviews(reviewsRes.data.reviews, id);
			this.props.addDetails(details);
			this.props.addReviews(reviews);
			this.setState({ loading: false, details, reviews: reviews.reviews });
		} catch (error) {
			this.setState({ error });
		}
	};

	render() {
		const { details, reviews } = this.state;
		return !this.state.loading ? (
			<div>
				<h1>{details.name}</h1>
				<img src={details.image} alt={details.name} />
				<p>{details.price}</p>
				<p>{details.address}</p>
				<p>{details.city}</p>
				<p>{details.phone}</p>
				<a href={details.yelpUrl} rel="noopener noreferrer" target="_blank">
					Website
				</a>
					<StarRatings rating={details.rating} starDimension="25px" starRatedColor="#af0000" />
				<ul>
					{reviews.map(r => <li>"{r}"</li>)}
				</ul>
				<p>Based on {details.reviewCount} review(s)</p>
			</div>
		) : (
			<div className="loader-container">
				<Loader message="Loading restaurant data..." />
			</div>
		);
	}
}

export const mapStateToProps = state => ({
	details: state.details,
	reviews: state.reviews
});

export const mapDispatchToProps = dispatch => ({
	addDetails: details => dispatch(addDetails(details)),
	addReviews: reviews => dispatch(addReviews(reviews))
});

export default connect(mapStateToProps, mapDispatchToProps)(RestaurantPage);
