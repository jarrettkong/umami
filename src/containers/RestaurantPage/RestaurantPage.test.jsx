import React from 'react';
import { shallow } from 'enzyme';
import { RestaurantPage } from './RestaurantPage';
import { mockDetails } from '../../util/mockData';
import { cleanDetails } from '../../util/cleaners';

jest.mock('../../util/cleaners');

describe('RestaurantPage', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = shallow(<RestaurantPage details={mockDetails} id={mockDetails[0].id} />);
	});

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	describe('componentDidMount', () => {
		it('should set the details to the state', () => {
			jest.spyOn(wrapper.instance(), 'getRestaurantDetails');
			wrapper.instance().componentDidMount();
			expect(wrapper.instance().getRestaurantDetails).not.toHaveBeenCalled();
			expect(wrapper.state('details')).toEqual(mockDetails[0]);
		});

		it('should call getRestaurantDetails if there is no existing info', () => {
			wrapper = shallow(<RestaurantPage details={[]} id={mockDetails[0].id} />);
			jest.spyOn(wrapper.instance(), 'getRestaurantDetails');
			wrapper.instance().componentDidMount();
			expect(wrapper.instance().getRestaurantDetails).toHaveBeenCalled();
		});
	});

	describe('getRestaurantDetails', () => {
		it.skip('should call Yelp.get with the correct params', () => {});

		it('should call cleanDetails with the raw data', async () => {
			await wrapper.instance().getRestaurantDetails(mockDetails[0].id);
			expect(cleanDetails).toHaveBeenCalledWith(mockDetails[0]);
		});
	});
});
