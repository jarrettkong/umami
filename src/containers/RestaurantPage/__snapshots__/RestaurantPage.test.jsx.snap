// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantPage should match the snapshot 1`] = `ShallowWrapper {}`;
